src/compiler/checker.ts

```diff
@@ -19747,9 +19747,10 @@ namespace ts {
                 // This ensures the subtype relationship is ordered, and preventing declaration order
                 // from deciding which type "wins" in union subtype reduction.
                 // They're still assignable to one another, since `readonly` doesn't affect assignability.
+                // This is only applied during the strictSubtypeRelation for compatability's sake in control flow

```

Comment:
It would probably be better to include which operations will use this, something like:

```suggestion
                // This is only applied during the strictSubtypeRelation -- currently used in subtype reduction
```